tests/integrations/resource_metering/test_cpu.rs

```diff
@@ -0,0 +1,277 @@
+        .rt
+        .spawn(require_cpu_time_not_zero(&test_suite, tag));
+
+    let table = ProductTable::new();
+    let insert = prepare_insert(&mut store, &table);
+    insert.execute();
+    store.commit();
+
+    let storage = store.get_storage();
+    let kvs = store.export();
+
+    let mut ctx = Context::default();
+    ctx.set_resource_group_tag(tag.as_bytes().to_vec());
+    let keys = kvs.iter().map(|(k, _)| k.as_slice()).collect::<Vec<_>>();
+    let res = storage.batch_get_command(ctx, &keys, u64::MAX).unwrap();
+
+    let expected_values = kvs.iter().map(|(_, v)| v);
+    let returned_values = res.iter().map(|kv| kv.as_ref().unwrap());
+    for (expected, returned) in expected_values.zip(returned_values) {
+        assert_eq!(expected, returned);
+    }
+    assert!(block_on(jh).unwrap());
+}
+
+fn setup_test_suite() -> (TestSuite, Store<RocksEngine>, Endpoint<RocksEngine>) {
+    let test_suite = TestSuite::new(resource_metering::Config {
+        report_receiver_interval: ReadableDuration::secs(3),
+        precision: ReadableDuration::secs(1),
+        ..Default::default()
+    });
+    let store = Store::from_storage(test_suite.get_storage());
+    tikv_util::thread_group::set_properties(Some(GroupProperties::default()));
+    let pool = ReadPool::from(readpool_impl::build_read_pool_for_test(
+        &CoprReadPoolConfig::default_for_test(),
+        store.get_engine(),
+    ));
+    let cm = ConcurrencyManager::new(1.into());
+    let endpoint = Endpoint::new(
+        &Default::default(),
+        pool.handle(),
+        cm,
+        PerfLevel::EnableCount,
+        test_suite.get_tag_factory(),
+    );
+    (test_suite, store, endpoint)
+}
+
+fn prepare_insert<'a>(
+    store: &'a mut Store<RocksEngine>,
+    table: &'a ProductTable,
+) -> Insert<'a, RocksEngine> {
+    store.begin();
+    store
+        .insert_into(table)
+        .set(&table["id"], Datum::I64(1))
+        .set(&table["name"], Datum::Bytes(b"name:0".to_vec()))
+        .set(&table["count"], Datum::I64(2))
+}
+
+fn require_cpu_time_not_zero(
+    test_suite: &TestSuite,
+    tag: &'static str,
+) -> impl Future<Output = bool> {
+    let (client, stream) = test_suite.subscribe();
+    async move {
+        let _client = client;
+        let tags = stream
+            .filter(|record| {
+                let record = record.as_ref().unwrap().clone();
+                async move { record.get_record().get_resource_group_tag() == tag.as_bytes() }
+            })
+            .take(1);
+        tags.collect::<Vec<_>>()
+            .await
+            .into_iter()
+            .map(|record| {
+                record
+                    .as_ref()
+                    .unwrap()
+                    .get_record()
+                    .get_items()
+                    .iter()
+                    .map(|item| item.cpu_time_ms)
+                    .sum::<u32>()
+            })
+            .sum::<u32>()
+            > 0
+    }
+}
+
+fn cpu_load() {
```

Comment:
How about using a Wall Time as a loop bound? In this way you can more precisely control how many CPU time it generates.

Reply:
good idea